shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -194,15 +194,18 @@ class JSBackend {
           case S(fnName) => fnName
           case N         => polyfill.use("withConstruct", topLevelScope.declareRuntimeSymbol("withConstruct"))
         }),
-        translateTerm(trm) :: JSRecord(fields map { case (Var(name), value) =>
+        translateTerm(trm) :: JSRecord(fields map { case (Var(name), (value, _)) =>
           name -> translateTerm(value)
         }) :: Nil
       )
     case Bra(_, trm) => translateTerm(trm)
     case Tup(terms) =>
-      JSArray(terms map { case (_, term) => translateTerm(term) })
+      JSArray(terms map { case (_, (term, _)) => translateTerm(term) })
     case Subs(arr, idx) =>
       JSMember(translateTerm(arr), translateTerm(idx))
+    case Assign(field, value) => JSImmEvalFn(N, Nil, 
```

Comment:
But what is the point of wrapping thins in a function?

Reply:
It seems that there's no need @_@